src/App.js

```diff
@@ -1,106 +1,134 @@
-import React from 'react';
+import React, { useRef, useState } from 'react';
 import './App.css';
 
 import firebase from 'firebase/app';
 import 'firebase/firestore';
 import 'firebase/auth';
+import 'firebase/analytics';
 
-import { useAuthState } from 'react-firebase-hooks/auth'
+import { useAuthState } from 'react-firebase-hooks/auth';
 import { useCollectionData } from 'react-firebase-hooks/firestore';
-
+require ('dotenv').config();
 
 firebase.initializeApp({
-  apiKey: "AIzaSyDEPYPHvbN3NnY3xeynCZumLmT4LCKpkiM",
-  authDomain: "frictionless-faction.firebaseapp.com",
-  projectId: "frictionless-faction",
-  storageBucket: "frictionless-faction.appspot.com",
-  messagingSenderId: "77867668377",
-  appId: "1:77867668377:web:c16c299d503b947f92f908",
-  measurementId: "G-SEYMGF0TT7"
+  apiKey: process.env.apiKey,
+  authDomain: process.env.authDomain,
+  projectId: process.env.projectId,
+  storageBucket: process.env.storageBucket,
+  messagingSenderId: process.env.messagingSenderId,
+  appId: process.env.appId,
+  measurementId: process.env.measurementId
+
 })
 
 
 
 
 const auth = firebase.auth();
 const firestore = firebase.firestore();
+const analytics = firebase.analytics();
 
 
+function App() {
 
-const [user] = useAuthState(auth);
+  const [user] = useAuthState(auth);
 
-function App() {
   return (
     <div className="App">
-      <header className="App-header">
-        <section>
-          {user ? <chatRoom /> : <signIn />}
-        </section>
+      <header>
+        <h1>⚛️🔥💬</h1>
+        <SignOut />
       </header>
+
+      <section>
+        {user ? <ChatRoom /> : <SignIn />}
+      </section>
+
     </div>
   );
 }
 
+function SignIn() {
 
-function signIn() {
   const signInWithGoogle = () => {
     const provider = new firebase.auth.GoogleAuthProvider();
     auth.signInWithPopup(provider);
-  };
+  }
 
-  return(
-    <button onClick = {signInWithGoogle}> Sign in with Google </button>
+  return (
+    <>
+      <button className="sign-in" onClick={signInWithGoogle}>Sign in with Google</button>
+      <p>Do not violate the community guidelines or you will be banned for life!</p>
+    </>
   )
 
-};
+}
 
-function signOut() {
+function SignOut() {
   return auth.currentUser && (
-    <button onClick = {() => auth.signOut()}> Sign Out</button>
+    <button className="sign-out" onClick={() => auth.signOut()}>Sign Out</button>
   )
+}
 
-};
 
+function ChatRoom() {
+  const dummy = useRef();
+  const messagesRef = firestore.collection('messages');
+  const query = messagesRef.orderBy('createdAt').limit(25);
 
+  const [messages] = useCollectionData(query, { idField: 'id' });
 
+  const [formValue, setFormValue] = useState('');
 
-function chatRoom() {
-  const messagesRef = firestore.collection('messages');
-  const query = messagesRef.orderBy('createdAt').limit(25);
 
-  const [messages] = useCollectionData(query, {idField: 'id'});
+  const sendMessage = async (e) => {
+    e.preventDefault();
```

Comment:
could you add a comment in the code explaining what `e` is? This will make documentation easier.

Reply:
I added comment line for it 

e as event